### PR TITLE
docs: adds depends_on

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Adds documentation about https://github.com/okteto/okteto/pull/3064

### CODE REVIEW

File: src/content/reference/manifest.mdx [hunk 1]
```diff
@@ -76,7 +76,7 @@ You can build all these images by running `okteto build`, or `okteto build xxx`
 
 > if a `<service>` name includes the symbol `-`, in the corresponding `OKTETO_BUILD_<service>` environment variables, the symbol `-` will be replaced by `_`.  For example, if `<service>` is `name-with-hyphen`, the environment variable names will start with `$OKTETO_BUILD_NAME_WITH_HYPHEN_`.
 
-> Okteto will automatically add all the build environment variables from previous builds on every build as a build argument. To refer to them remember to add the `ARG` instruction on your Dockerfile.
+> Okteto will automatically add all the build environment variables from previous builds as a build argument. To refer to them, remember to add the `ARG` instruction on your Dockerfile.
```
Comment (RinkiyaKeDad) [inline src/content/reference/manifest.mdx:79]:
```suggestion
> Okteto will automatically add all the build environment variables from previous builds as build arguments. To refer to them, remember to add the `ARG` instruction on your Dockerfile.
```

**Review by RinkiyaKeDad — APPROVED** [review]

File: src/content/reference/manifest.mdx [hunk 2]
```diff
@@ -60,6 +63,7 @@ Each image supports the following fields:
 - `args`: add build arguments, which are environment variables accessible only during the build process. Build arguments with a value containing a `$` sign are resolved to the environment variable value on the machine okteto is running on.
 - `export_cache`: image tag for exported cache when build.
 - `cache_from`: list of images to import cache from.
+- `depends_on`: list of images that need to be built first.
```
Comment (rberrelleza) [inline src/content/reference/manifest.mdx:66]: What happens if there's a dependency cycle? Do we detect it? (Image A depends on B, and image B depends on A)

Comment (jLopezbarb) [reply src/content/reference/manifest.mdx:66]: Sorry for the late response. Yes, we detect it and throw a proper error

**Comment by jLopezbarb** [comment]
We decided not to merge this one until we release https://github.com/okteto/okteto/pull/3115 which will be included in 2.8. I also need to update the documentation with the changes from that PR

File: src/content/reference/manifest.mdx [hunk 3]
```diff
@@ -72,6 +78,8 @@ You can build all these images by running `okteto build`, or `okteto build xxx`
 
 > if a `<service>` name includes the symbol `-`, in the corresponding `OKTETO_BUILD_<service>` environment variables, the symbol `-` will be replaced by `_`.  For example, if `<service>` is `name-with-hyphen`, the environment variable names will start with `$OKTETO_BUILD_NAME_WITH_HYPHEN_`.
 
+> Okteto will automatically add all the build environment variables from previous builds as build arguments. To refer to them, remember to add the `ARG` instruction on your Dockerfile.
```
Comment (rberrelleza) [inline src/content/reference/manifest.mdx:81]: What do we mean by "previous builds"?

Comment (jLopezbarb) [reply src/content/reference/manifest.mdx:81]: The builds of the services that has been already build (the ones that are under `depends_on` section)

Comment (rberrelleza) [inline src/content/reference/manifest.mdx:81]:
```suggestion
> Okteto will automatically add all the build environment variables from all previous images in the dependency chain as build arguments. To refer to them, remember to add the `ARG` instruction on your Dockerfile.
```

**Review by rberrelleza — APPROVED** [review]

**Pull request merged**